File: flutterapp/android/app/src/main/kotlin/de/kneke/flutterapp/channel/BaseResourceChannel.kt
package de.kneke.flutterapp.channel

import de.kneke.flutterapp.channel.ChannelConfig
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel

abstract class BaseResourceChannel(channelConfig: ChannelConfig, flutterEngine: FlutterEngine) {

    init {
        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, channelConfig.getMethodChannelName())
            .setMethodCallHandler { call, result ->
                when (call.method) {
                    channelConfig.getChannelMethods()[0] -> get(call.arguments, result)
                    else -> result.notImplemented()
                }
            }
    }

    abstract fun get(arguments: Any?, result: MethodChannel.Result)
}

File: flutterapp/ios/Runner/channel/BaseResourceChannel.swift

import Foundation
import Flutter

class BaseResourceChannel {
    
    let channelConfig: ChannelConfig
    let flutterController: FlutterViewController
    
    init(channelConfig: ChannelConfig, flutterController: FlutterViewController) {
        self.channelConfig = channelConfig
        self.flutterController = flutterController
    }
    
    func regiterChannels() {
        let methodChannel = FlutterMethodChannel(name: channelConfig.getMethodChannelName(), binaryMessenger: flutterController.binaryMessenger)
        methodChannel.setMethodCallHandler({
            [self] (call: FlutterMethodCall, result: FlutterResult) -> Void in
            switch call.method {
            case self.channelConfig.getChannelMethods()[0]:
                self.get(arguments: call.arguments, result: result)
            default:
                print(call.method)
                print(self.channelConfig.getChannelMethods()[0])
                result(FlutterMethodNotImplemented)
            }
        })
    }
    
    func get(arguments: Any?, result: FlutterResult) {
        fatalError("Must be implemented in Subclass")
    }
}
